Lib/fontbakery/profiles/name.py

```diff
@@ -592,6 +592,80 @@ def com_adobe_fonts_check_family_max_4_fonts_per_family_name(ttFonts):
         yield PASS, ("There were no more than 4 fonts per family name.")
 
 
+
+@check(
+    id = 'com.adobe.fonts/check/family/consistent_family_name',
+    rationale = """
+        Per the OpenType spec:
+            * "...many existing applications that use this pair of names assume that a
+              Font Family name is shared by at most four fonts that form a font
+              style-linking group"
+            * "For extended typographic families that includes fonts other than the 
+              four basic styles(regular, italic, bold, bold italic), it is strongly
+              recommended that name IDs 16 and 17 be used in fonts to create an extended,
+              typographic grouping."
+            * "If name ID 16 is absent, then name ID 1 is considered to be the typographic
+               family name."
+
+        https://learn.microsoft.com/en-us/typography/opentype/spec/name
+
+        Fonts within a font family all must have consistent names 
+        in the Typographic Family name (nameID 16)
+        or Font Family name (nameID 1), depending on which it uses.
+
+        Inconsistent font/typographic family names across fonts in a family
+        can result in unexpected behaviors, such as broken style linking.
+    """,
+)
+def com_adobe_fonts_check_consistent_font_family_name(ttFonts):
+    """Verify that family names in the name table are consistent across all fonts in the family.
+       Checks Typographic Family name (nameID 16) if present, 
+       otherwise uses Font Family name (nameID 1)
+    """
+    from fontbakery.utils import get_name_entry_strings
+    from collections import defaultdict
+    import os
+
+    name_dict = defaultdict(list)
+    for ttFont in ttFonts:
+        filename = os.path.basename(ttFont.reader.file.name)
+        # try nameID 16
+        nameID = NameID.TYPOGRAPHIC_FAMILY_NAME
+        names_list = get_name_entry_strings(ttFont,
+                                            NameID.TYPOGRAPHIC_FAMILY_NAME,
+                                            PlatformID.WINDOWS,
+                                            WindowsEncodingID.UNICODE_BMP,
+                                            WindowsLanguageID.ENGLISH_USA)
+        if len(names_list) == 0:
+            # use nameID 1 instead
+            nameID = NameID.FONT_FAMILY_NAME
+            names_list = get_name_entry_strings(ttFont,
+                                                NameID.FONT_FAMILY_NAME,
+                                                PlatformID.WINDOWS,
+                                                WindowsEncodingID.UNICODE_BMP,
+                                                WindowsLanguageID.ENGLISH_USA)
+        name_dict[frozenset(names_list)].append((filename, nameID))
+
+    if len(name_dict) > 1:
+        detail_str_arr = []
+        indent = "\n  - "
+        for name_set, font_tuple_list in name_dict.items():
+            detail_str = f"\n\n* {next(iter(name_set))!r} was found in:"
```

Comment:
This line is causing a pylint error in Ubuntu 3.9. It seems suspicious that it's the only environment that fails. I'm looking into it.

Reply:
@kaydeearts please change `next(iter(name_set))` to `next(iter(name_set), '')`.

If the CI still fails after that change, append this comment to the line instead

      # pylint: disable=stop-iteration-return